File: 566-reshape-the-matrix/566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        if len(mat) * len(mat[0]) != r*c: 
            return mat
        result = []
        tempRow = []
        i = 0
        for row in mat:
            for element in row:
                tempRow.append(element)
                i+=1
                if i == c:
                    i=0
                    result.append(tempRow)
                    tempRow = []
        return result
        

File: 566-reshape-the-matrix/566-reshape-the-matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int row = mat.length;
        int col = mat[0].length;
        if(row*col != r*c){
            return mat;
        }
        
        int result[][] = new int[r][c];
        int x = 0;
        int y = 0;
        for(int i = 0; i < r;i++){
            for(int j = 0; j < c; j++){
                if(y == col){
                    x++;
                    y = 0;
                }
                result[i][j] = mat[x][y];
                y++;
            }
        }
        return result;
        
        
    }
}